packages/flutter/lib/src/material/carousel.dart

```diff
@@ -1578,6 +1747,10 @@ class CarouselController extends ScrollController {
   /// The item that expands to the maximum size when first creating the [CarouselView].
   final int initialItem;
 
+  /// The current index of the carousel.
```

Comment:
We might need more explanation about what the "current" index stands for.

For `initialItem`, if we create a CarouselView(), it means the first item in the view; if we create a CarouselView.weight(), it means the first item with the max weight in the view. So if the `currentIndex` defaults to `initialItem`, we need to match the behavior.